crates/subspace-networking/src/node_runner.rs

```diff
@@ -348,6 +348,16 @@ where
                 num_established,
                 ..
             } => {
+                // Save known addresses that were successfully dialed.
+                if let ConnectedPoint::Dialer { address, .. } = &endpoint {
+                    // filter non-global addresses when non-globals addresses are disabled
+                    if is_global_address_or_dns(address) || self.allow_non_global_addresses_in_dht {
+                        self.networking_parameters_registry
+                            .add_known_peer(peer_id, vec![address.clone()])
```

Comment:
This is where I see `add_known_peer` is not idiomatic. You clone a data structure, then allocate a vector, all without actually needing vector in` add_known_peer` implementation. It should have accepted `&[Multiaddr]` as argument, then you can create that from `&Multiaddr` using https://doc.rust-lang.org/stable/std/slice/fn.from_ref.html

Reply:
This change requires an introduction of lifetimes which I prefer to avoid here.